### PR TITLE
keyless bundling: verify the cert!

### PR DESCRIPTION
**Pull request by priyawadhwa**
To verify keyless mode we need the integrated time for the log entry to verify the cert is valid. So, even though I  literally just decided to store the canonicailzed payload in the bundle in the last PR, I switched to storing the body, integrated time and log index in the bundle in this PR

This way we can generate the canonicalized payload ourselves, which verifies that the integrated time provided in the annotation is correct. Then we can use that integrated time to verify the cert!

Output looks like:

```
Verification for gcr.io/priya-wadhwa/test --
The following checks were performed on each of these signatures:
  - The specified annotations were verified.
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":"gcr.io/priya-wadhwa/test"},"image":{"docker-manifest-digest":"sha256:dfa3cef088454200d6b48e2a911138f7d5d9afff77f89243eea6342f16ddcfb0"},"type":"cosign container image signature"},"optional":{"CommonName":"priyawadhwa@google.com","yay":"yay"}}]
```

Signed-off-by: Priya Wadhwa <priyawadhwa@google.com>

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -105,13 +105,17 @@ Gyp4apdU7AXEwysEQIb034aPrTlpmxh90SnTZFs2DHOvCjCPPAmoWfuQUwPhSPRb
...
...FlVkl2Y1FvNU1FSnNOM2hzUm5OaVNVWTVTWEZCUW1ReEwwRkpWV0pqT0RScVkzbHZOa1Z6U21aRmFYZFRabkIxY2s5Wk5IbHFWVEJSV2s4dloyODBTVUpMYWtORENrRlRXWGRFWjFsRVZsSXdVRUZSU0M5Q1FWRkVRV2RsUVUxQ1RVZEJNVlZrU2xGUlRVMUJiMGREUTNOSFFWRlZSa0ozVFVSTlFYZEhRVEZWWkVWM1JVSUtMM2RSUTAxQlFYZElVVmxFVmxJd1QwSkNXVVZHUzNkeGVtVnpNbGRQU1M4MFEycFVkMnRLWW5Wck0wRmthMDVzVFVJNFIwRXhWV1JKZDFGWlRVSmhRUXBHVFdwR1NGRkNRbTFwVVhCTmJFVnJObmN5ZFZOMU1VdENkRkJ6VFVsSFRrSm5aM0pDWjBWR1FsRmpRa0ZSVTBKblJFSXJUVWgzUjBORGMwZEJVVlZHQ2tKNlFVTm9ia0p2WkVoU2QwOXBPSFpqU0Vwd1pHMUdNRnBYVG1oTVYwNTJZbTVTYkdKdVVYUk9ha0Y2V20xVk0xcFVZM1JOUkVGM1RVTXdlVTFxU1RNS1RGZEtiVTU2VlhSYWFsSnRUbGRWTkUxSFVYbFBWRlV3VEc1T01HSXpTbWhhTWxWMVdqSTVkbG95ZUd4WldFSndZM2sxYW1JeU1IWlpNa1Y2VG0xRmVBcGFWR3N5VFdwUmVWbHFiRzFaTWtsNFRrUlpkbGt5UlhWWk0wb3dUVU5GUjBFeFZXUkZVVkZoVFVKcFFrWnVRbmxoV0d4b1pESkdhMkZJWkdoUlIyUjJDbUl5WkhOYVV6VnFZakl3ZDBObldVbExiMXBKZW1vd1JVRjNUVVJhZDBGM1drRkpkMGhqUjBWVFprMUtPRk41UmsxWk1GZHJhMHd4Wm1aNVdHcERlQ3NLVVZoUVRUZFFZbkEyZDJWWFpYUmtabU4zUzBoelRFNU1NMDFTTTJwbGFXeG9iV0p2UVdwQmVVcGtiamxQVm5sVllrRkJPVXR0VUdoa1kwcDVTRlIwV1Fwb1NFZzBObXBzUjJNMFJXMURSVXBwU2sxNmVuWm5WamRSYmxOeU5HaFJlRWRoYjFWQlowRTlDaTB0TFMwdFJVNUVJRU5GVWxSSlJrbERRVlJGTFMwdExTMEsifX19LCJraW5kIjoicmVrb3JkIn0=",
```
Comment (dlorenc) [inline SPEC.md:109]: LBTM

File: SPEC.md [hunk 2]
```diff
@@ -105,13 +105,17 @@ Gyp4apdU7AXEwysEQIb034aPrTlpmxh90SnTZFs2DHOvCjCPPAmoWfuQUwPhSPRb
...
+  "IntegratedTime": 1619819449,
+  "LogIndex": 3460
 }
 ```
   The following are REQUIRED properties of the bundle:
     - The `SignedEntryTimestamp` is a rekor-signed signature over the logIndex, body and integratedTime of the Rekor Log Entry
-    - The `CanonicalizedPayload` is a JSON document following RFC 8785 rules consisting of the logIndex, body and integratedTime of the Rekor Log Entry
+    - The `Body` is the body of the Rekor Log Entry
+    - The `IntegratedTime` is the time the log entry was integrated into the transparency log
```
Comment (dlorenc) [inline SPEC.md:117]: Specify the format - I think it's just a UNIX timestamp right?

**Review by dlorenc — COMMENTED** [review]
One nit.

File: SPEC.md (hunk 1 above)
Comment (priyawadhwa) [reply SPEC.md:109]: fr tho 😞

File: SPEC.md (hunk 2 above)
Comment (priyawadhwa) [reply SPEC.md]: 👍🏽

File: SPEC.md (hunk 1 above)
Comment (dlorenc) [reply SPEC.md:109]: :)

**Review by dlorenc — APPROVED** [review]

**Pull request merged**